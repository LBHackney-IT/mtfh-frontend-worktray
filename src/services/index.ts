export { default as locale } from "./locale";
export { config } from "./config";
