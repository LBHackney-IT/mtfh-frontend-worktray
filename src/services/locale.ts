const locale = {
  title: "Your tasks",
  views: {
    worktray: {},
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  components: {
    filters: {
      applyFilters: "Apply filters",
      clearFilters: "Clear filters",
      selectAll: "Select All",
    },
  },
};

export default locale;
