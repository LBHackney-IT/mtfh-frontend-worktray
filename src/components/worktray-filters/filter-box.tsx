import React from "react";

import { Checkbox, Text } from "@mtfh/common/lib/components";

import { locale } from "../../services";

import "./styles.scss";

const { components } = locale;

export const FilterBox = ({
  filterType,
  title,
  options,
  handleSelectAll,
  handleCheckboxFilters,
  selectedFilters,
}: {
  filterType: string;
  title: string;
  options: string[];
  handleSelectAll: (value: string) => void;
  handleCheckboxFilters: (
    value: React.ChangeEvent<HTMLInputElement>,
    filterType: string,
  ) => void;
  selectedFilters: Record<string, string[]>;
}): JSX.Element => {
  return (
    <div className="filter-box">
      <div className="filter-header">
        <Text>{title}</Text>
        <button
          className="select-all lbh-link lbh-link--no-visited-state"
          onClick={() => handleSelectAll(filterType)}
        >
          {components.filters.selectAll}
        </button>
      </div>

      <div className="filter-box-options">
        {options.map((option) => {
          return (
            <Checkbox
              key={option.toLowerCase().replace(/\s/g, "-")}
              id={option.toLowerCase().replace(/\s/g, "-")}
              name={option}
              checked={selectedFilters[filterType].includes(option)}
              onChange={(event) => handleCheckboxFilters(event, filterType)}
            >
              {option}
            </Checkbox>
          );
        })}
      </div>
    </div>
  );
};
