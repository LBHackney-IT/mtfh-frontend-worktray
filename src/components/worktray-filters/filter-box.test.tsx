import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { fireEvent, screen } from "@testing-library/react";

import { locale } from "../../services";
import { FilterBox } from "./filter-box";

const handleSelectAll = jest.fn();
const handleRemoveAll = jest.fn();
const handleCheckboxFilters = jest.fn();

const { components } = locale;

describe("filter-box-component", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  test("it renders FilterBox correctly", async () => {
    const filter = {
      type: "processes",
      title: "Processes",
      options: [
        { key: "process-1", value: "Process 1" },
        { key: "process-2", value: "Process 2" },
        { key: "process-3", value: "Process 3" },
      ],
    };
    const { container } = render(
      <FilterBox
        filterType={filter.type}
        title={filter.title}
        options={filter.options}
        handleSelectAll={handleSelectAll}
        handleRemoveAll={handleRemoveAll}
        handleCheckboxFilters={handleCheckboxFilters}
        selectedFilters={{ [filter.type]: [] }}
      />,
    );

    await expect(screen.findByText(filter.title)).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(components.filters.selectAll),
    ).resolves.toBeInTheDocument();
    fireEvent.click(screen.getByText(components.filters.selectAll));
    expect(handleSelectAll.mock.calls.length).toBe(1);
    await expect(screen.findByText(filter.options[0].value)).resolves.toBeInTheDocument();
    await expect(screen.findByText(filter.options[1].value)).resolves.toBeInTheDocument();
    await expect(screen.findByText(filter.options[2].value)).resolves.toBeInTheDocument();
    fireEvent.click(screen.getByText(filter.options[0].value));
    expect(handleCheckboxFilters.mock.calls.length).toBe(1);
    expect(container).toMatchSnapshot();
  });

  test("it renders FilterBox with remove all correctly", async () => {
    const filter = {
      type: "processes",
      title: "Processes",
      options: [{ key: "process-1", value: "Process 1" }],
    };
    render(
      <FilterBox
        filterType={filter.type}
        title={filter.title}
        options={filter.options}
        handleSelectAll={handleSelectAll}
        handleRemoveAll={handleRemoveAll}
        handleCheckboxFilters={handleCheckboxFilters}
        selectedFilters={{ [filter.type]: ["process-1"] }}
      />,
    );

    await expect(screen.findByText(filter.title)).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(components.filters.removeAll),
    ).resolves.toBeInTheDocument();
    fireEvent.click(screen.getByText(components.filters.removeAll));
    expect(handleRemoveAll.mock.calls.length).toBe(1);
  });
});
