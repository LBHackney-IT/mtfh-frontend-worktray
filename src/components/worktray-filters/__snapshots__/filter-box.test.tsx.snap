// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-box-component it renders FilterBox correctly 1`] = `
<div>
  <div
    class="filter-box"
  >
    <div
      class="filter-header"
    >
      <p
        class="lbh-body-m"
      >
        Processes
      </p>
      <button
        class="select-all lbh-link lbh-link--no-visited-state"
      >
        Select All
      </button>
    </div>
    <div
      class="filter-box-options"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="process-1"
          name="Process 1"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="process-1"
        >
          Process 1
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="process-2"
          name="Process 2"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="process-2"
        >
          Process 2
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="process-3"
          name="Process 3"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="process-3"
        >
          Process 3
        </label>
      </div>
    </div>
  </div>
</div>
`;
